lib/model/avatar_url.dart

```diff
@@ -0,0 +1,138 @@
+import '../log.dart';
+
+/// Pixel dimensions of different size choices we have (they're all
+/// square) when requesting an uploaded avatar.
+/// DEFAULT_AVATAR_SIZE in zerver/lib/upload.py.
+const int defaultUploadSizePx = 100;
+
+/// Pixel dimensions of different size choices we have (they're all
+/// square) when requesting an uploaded avatar.
+/// MEDIUM_AVATAR_SIZE in zerver/lib/upload.py.
+const int mediumUploadSizePx = 500;
+
+abstract class AvatarUrl {
+  static AvatarUrl fromUserData({
+    required String rawAvatarUrl,
+    required Uri realmUrl,
+    required int userId,
+  }) {
+    // TODO(#255): handle computing gravatar's
+    // TODO(#254): handle computing Fallback avatar
+    if (rawAvatarUrl.startsWith(GravatarUrl.origin)) {
+      return GravatarUrl(
+        urlFromServer: rawAvatarUrl);
+    } else {
+      return UploadedAvatarUrl(
+        realmUrl: realmUrl, absoluteOrRelativeUrl: rawAvatarUrl);
+    }
+  }
+
+  Uri get(double sizePhysicalPx);
+}
+
+class GravatarUrl implements AvatarUrl {
+  GravatarUrl._({required this.standardUri});
+
+  factory GravatarUrl({
```

Comment:
nit: put this private constructor after the public one

(this public one is also the "default" constructor, i.e. the one with no name beyond the class's name — so that's the constructor the Flutter style guide I linked previously says to put first)

Reply:
https://github.com/zulip/zulip-flutter/pull/534/commits/3230320ec22a46485e6d1c8c7829f3c5207aef5e